src/main/java/com/github/manolo8/darkbot/utils/login/LoginUtils.java

```diff
@@ -118,17 +117,37 @@ public static void usernameLogin(LoginData loginData) {
     }
 
     public static void usernameLogin(LoginData loginData, String domain) {
-        String loginUrl = Http.create("https://" + domain + ".darkorbit.com/")
-                .consumeInputStream(LoginUtils::getLoginUrl);
+        String frontPage = "";
+        String gResponse = "";
+        try {
+            frontPage = IOUtils.read(Http.create("https://" + domain + ".darkorbit.com/").getInputStream());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
 
+        try {
+            CaptchaAPI captcha = CaptchaAPI.getInstance();
+            gResponse = captcha.solveCaptcha(frontPage);
+        }catch (Exception e){
+            e.printStackTrace();
+        }
+
+        String loginUrl = getLoginUrl(frontPage);
         CookieManager cookieManager = new CookieManager();
         CookieHandler.setDefault(cookieManager);
 
         try {
-            Http.create(loginUrl, Method.POST)
-                    .setParam("username", loginData.getUsername())
-                    .setParam("password", loginData.getPassword())
-                    .closeInputStream();
+            if(gResponse.isEmpty())
+                Http.create(loginUrl, Method.POST)
+                        .setParam("username", loginData.getUsername())
+                        .setParam("password", loginData.getPassword())
+                        .closeInputStream();
+            else
+                Http.create(loginUrl, Method.POST)
+                        .setParam("username", loginData.getUsername())
+                        .setParam("password", loginData.getPassword())
+                        .setParam("g-recaptcha-response", gResponse)
+                        .closeInputStream();
```

Comment:
```suggestion
            Http http = Http.create(loginUrl, Method.POST)
                        .setParam("username", loginData.getUsername())
                        .setParam("password", loginData.getPassword());
            extraPostParams.forEach(http::setParam);
            http.closeInputStream();
```

Simply add whatever extra params exist directly, it could be 0, or many, not relevant to our implementation